tools/cmock/create_test.cmake

```diff
@@ -0,0 +1,168 @@
+# Taken from amazon-freertos repository
+
+#function to create the test executable
+function(create_test test_name
+                     test_src
+                     link_list
+                     dep_list
+                     include_list)
+    set(mocks_dir "${CMAKE_CURRENT_BINARY_DIR}/mocks")
+    include (CTest)
+    get_filename_component(test_src_absolute ${test_src} ABSOLUTE)
+    add_custom_command(OUTPUT ${test_name}_runner.c
+                  COMMAND ruby
+                    ${CMOCK_DIR}/vendor/unity/auto/generate_test_runner.rb
+                    ${MODULE_ROOT_DIR}/tools/cmock/project.yml
+                    ${test_src_absolute}
+                    ${test_name}_runner.c
+                  DEPENDS ${test_src}
+        )
+    add_executable(${test_name} ${test_src} ${test_name}_runner.c)
+    set_target_properties(${test_name} PROPERTIES
+            COMPILE_FLAG "-O0 -ggdb"
+            RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin/tests"
+            INSTALL_RPATH_USE_LINK_PATH TRUE
+            LINK_FLAGS " \
+                -Wl,-rpath,${CMAKE_BINARY_DIR}/lib \
+                -Wl,-rpath,${CMAKE_CURRENT_BINARY_DIR}/lib"
+        )
+    target_include_directories(${test_name} PUBLIC
+                               ${mocks_dir}
+                               ${include_list}
+        )
+
+    target_link_directories(${test_name} PUBLIC
+                            ${CMAKE_CURRENT_BINARY_DIR}
+        )
+
+    # link all libraries sent through parameters
+    foreach(link IN LISTS link_list)
+        target_link_libraries(${test_name} ${link})
+    endforeach()
+
+    # add dependency to all the dep_list parameter
+    foreach(dependency IN LISTS dep_list)
+        add_dependencies(${test_name} ${dependency})
+        target_link_libraries(${test_name} ${dependency})
+    endforeach()
+    target_link_libraries(${test_name} -lgcov unity)
+    target_link_directories(${test_name}  PUBLIC
+                            ${CMAKE_CURRENT_BINARY_DIR}/lib
+            )
+    add_test(NAME ${test_name}
+             COMMAND ${CMAKE_BINARY_DIR}/bin/tests/${test_name}
+             WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+            )
+endfunction()
+
+# Run the C preprocessor on target files.
+# Takes a CMAKE list of arguments to pass to the C compiler
+function(preprocess_mock_list mock_name file_list compiler_args)
+    set_property(GLOBAL PROPERTY ${mock_name}_processed TRUE)
+    foreach (target_file IN LISTS file_list)
+        # Has to be TARGET ALL so the file is pre-processed before CMOCK
+        # is executed on the file.
+        add_custom_command(OUTPUT ${target_file}.backup
+            COMMAND scp ${target_file} ${target_file}.backup
+            VERBATIM COMMAND ${CMAKE_C_COMPILER} -E ${compiler_args} ${target_file} > ${target_file}.out
```

Comment:
Typically preprocessed files have the a .i extension.
You could probably use -o rather than > to handle the output.

Reply:
I referred to this from [coreMQTT ](https://github.com/FreeRTOS/coreMQTT/blob/f756d9833ad79b60084def59935fdabf475c08b0/tools/cmock/create_test.cmake#L67).
Does this not correct?

Reply:
It is correct either way.

Reply:
Thanks.